test/test-authorizedRequestFactory.js

```diff
@@ -0,0 +1,93 @@
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+var proxyquire = require('proxyquire').noPreserveCache();
+var nock = require('nock');
+var assert = require('assert');
+
+nock.disableNetConnect();
+describe('utils.authorizedRequestFactory', function() {
+  var authMock = nock('https://accounts.google.com:443')
+  .persist()
+  .post('/o/oauth2/token')
+  .reply(200, {
+    access_token: 'stub_token',
+    token_type: 'Bearer',
+    expires_in: 3600
+  });
+  after(function() {
+    nock.cleanAll();
+    nock.enableNetConnect();
+  });
+ 
+   describe('authorizedRequestFactory', function() {
+    it('should return an error on broken configuration', function (done) {
+      var oldEnv = process.env.GOOGLE_APPLICATION_CREDENTIALS;
+      process.env.GOOGLE_APPLICATION_CREDENTIALS = './not-a-file.json';
+      var utils = require('../lib/utils.js');
+      var req = utils.authorizedRequestFactory(['https://www.googleapis.com/auth/cloud-platform']);
+      req({'X-custom-header': 'true', url: 'http://www.test.com/test'}, 
+        function (err, response, body) {
+          assert.ok(err instanceof Error, 'Should be an error');
+          process.env.GOOGLE_APPLICATION_CREDENTIALS = oldEnv;
+          done();
+        }
+      );
+    });
+    it('should return a function', function() {
+      var utils = require('../lib/utils.js');
+      var result = utils.authorizedRequestFactory(['https://www.googleapis.com/auth/cloud-platform']);
+      assert(typeof result === 'function');
+    });
+    it('should be able to make a valid request with options', function (done) {
+      var utils = require('../lib/utils.js');
+      var req = utils.authorizedRequestFactory(['https://www.googleapis.com/auth/cloud-platform']);
+      var mock = nock('http://www.test.com')
+        .persist()
+        .get('/test')
+        .reply(200, 'test');
+      req({'X-custom-header': 'true', url: 'http://www.test.com/test'}, 
+        function (err, response, body) {
+          console.log('got error', err)
+          assert.deepEqual(err, null, 'error should be null');
+          assert.ok(typeof response === 'object');
+          assert.deepEqual(body, 'test');
+          mock.done();
+          nock.cleanAll();
+          done();
+        }
+      );
+    });
+    it('should be able to make a valid request with no options', function (done) {
+      var utils = require('../lib/utils.js');
+      var req = utils.authorizedRequestFactory(['https://www.googleapis.com/auth/cloud-platform']);
+      var mock = nock('http://www.test.com')
+        .persist()
```

Comment:
I don't think so since the retry test functionality is unaltered and this tests a success case where a retry is not needed but I am not sure if that is still a candidate for further testing down the road.